ci/tests/general/file_reader.py

```diff
@@ -0,0 +1,448 @@
+                    if count > self.LIMIT:
+                        raise IndexError("ERROR: ITERATION LIMIT EXCEEDED")
+                if debug:
+                    print("...stopping: '{0}'".format(self.line))
+                # Rewind a bit to make sure we didn't miss any
+                seek = old_mid_range
+                while self.line_date >= search_start and seek > 0:
+                    if seek < self.REWIND:
+                        seek = 0
+                    else:
+                        seek -= self.REWIND
+                    if debug:
+                        print("...rewinding")
+                    self.opened_file.seek(seek)
+                    # sync to self.line ending
+                    self.line_date, self.line = self._read_line()
+                    if debug:
+                        print("...junk: '{0}'".format(self.line))
+                    self.line_date, self.line = self._read_line()
+                    if debug:
+                        print("...comparing: '{0}'".format(self.line_date))
+
+                # Scan forward
+                while self.line_date < search_start:
+                    if debug:
+                        print("...skipping: '{0}'".format(self.line_date))
+                    self.line_date, self.line = self._read_line()
+                if debug:
+                    print("...found: '{0}'".format(self.line))
+                if debug:
+                    print("Beginning: {0} Mid: {1} End: {2} P1: {3} P2: {4} Timestamp: '{5}'".
+                          format(begin_range, mid_range, end_range, pos1, pos2, self.line_date))
+                # Now that the preliminaries are out of the way, we just loop,
+                # reading lines and printing them until they are beyond the end of the range we want
+                while self.line_date <= search_end:
+                    # Search for the search patterns before printing
+                    if search_patterns:
+                        if any(pattern.lower() in self.line.lower() for pattern in search_patterns):
+                            # Write to file to save memory and finally return its contents
+                            with open(self.FILE_PATH, 'a') as f:
+                                f.write(str(write_prefix + LogFileTimeParser._clean_text(self.line) + '\n'))
+                    else:
+                        # Write to file to save memory and finally return its contents
+                        with open(self.FILE_PATH, 'a') as f:
+                            f.write(str(write_prefix + LogFileTimeParser._clean_text(self.line) + '\n'))
+                    self.line_date, self.line = self._read_line()
+                if debug:
+                    print("Start: '{0}' End: '{1}'".format(search_start, search_end))
+                # Done reading from file
+                self.opened_file.close()
+            # Do not display EOFErrors:
+            except EOFError:
+                pass
```

Comment:
It actually would. By passing on that exception, the statement above would be reached and it would close.